Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM golang:1.12 as build
+
+WORKDIR /iavl
+
+ARG GOFLAGS=""
+ENV GOFLAGS=$GOFLAGS
+ENV GO111MODULE=on
+
+# Download dependencies first - this should be cacheable.
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Now add the local iavl repo, which typically isn't cacheable.
+COPY . .
+
+# Build the server.
+RUN go get ./cmd/iavlserver
+
+# Make a minimal image.
+FROM gcr.io/distroless/base
+
+COPY --from=build /go/bin/iavlserver /
```

Comment:
This should have a default entrypoint or command. The typical way to do this is to use an entrypoint with the base binary, and a default command giving some basic parameters for a transient database. We should also expose the port numbers.

Reply:
I've been running with something like

```
> sudo docker run --name iavl -p 8090:8090 -p 8091:8091 -d martin/iavl:latest /iavlserver  -db-name "test" -datadir "."  -grpc-endpoint "0.0.0.0:8090"  -gateway-endpoint "0.0.0.0:8091"

```

I'm not like a Dockerfile power user, but are you saying you want to use environment variables instead of this kind of invocation to pass the cli args and flags?

Reply:
No, I'm saying we should start `/iavlserver` with a transient database by default:

```Dockerfile
EXPOSE 8090 8091
ENTRYPOINT ["/iavlserver"]
COMMAND ["-db-name", "iavl", "-datadir", "."]
```

Users can override the parameters like so:

```sh
$ docker run -d iavl:latest -db-name "foo"
```